python/res/job_queue/job_queue_manager.py

```diff
@@ -145,16 +124,28 @@ def __repr__(self):
     def max_running(self):
         return self.get_job_queue().get_max_running()
 
+    def max_submit(self):
+        return self.get_job_queue().get_max_submit()
+
     def execute_queue(self):
 
         job_queue = self.get_job_queue()
         started_job_threads = []
         while job_queue.is_running():
             job = job_queue.fetch_next_waiting()
             while not job_queue.stopped_by_user and job is not None and job_queue.count_running() <= self.max_running():
-                started_job_threads.append(job.run(job_queue.driver))
+                started_job_threads.append(job.run(job_queue.driver, max_submit=self.max_submit()))
                 job = job_queue.fetch_next_waiting()
             time.sleep(1)
+            if job_queue.stopped_by_user:
+                for job in job_queue.job_list:
+                    job.stop(job_queue.driver)
 
         for thread in started_job_threads:
             thread.join()
+
+        #Cleanup all not-started jobs
+        for job in job_queue.job_list:
+            if job.status == JobStatusType.JOB_QUEUE_WAITING:
+                job._set_status(JobStatusType.JOB_QUEUE_IS_KILLED)
```

Comment:
It looks like this PR includes some of the same changes in #757 
I guess you and @ManInFez should fight over it ;)

Reply:
That was agreed on standup before @ManInFez went for vacation.